src/pages/EnablePayments/AddBankAccount/AddBankAccount.tsx

```diff
@@ -0,0 +1,137 @@
+import * as PaymentMethods from '@userActions/PaymentMethods';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type {PersonalBankAccountForm} from '@src/types/form';
+import type {PersonalBankAccount, PlaidData, UserWallet} from '@src/types/onyx';
+
+type AddPersonalBankAccountPageWithOnyxProps = {
+    /** The details about the Personal bank account we are adding saved in Onyx */
+    personalBankAccount: OnyxEntry<PersonalBankAccount>;
+
+    /** The user's wallet */
+    userWallet: OnyxEntry<UserWallet>;
+
+    /** Contains plaid data */
+    plaidData: OnyxEntry<PlaidData>;
+
+    /** The draft values of the bank account being setup */
+    personalBankAccountDraft: OnyxEntry<PersonalBankAccountForm>;
+};
+
+const plaidSubsteps: Array<React.ComponentType<SubStepProps>> = [Plaid, Confirmation];
+
+function AddBankAccount({personalBankAccount, userWallet, plaidData, personalBankAccountDraft}: AddPersonalBankAccountPageWithOnyxProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+
+    const submit = useCallback(() => {
+        const bankAccounts = plaidData?.bankAccounts ?? [];
+        const selectedPlaidBankAccount = bankAccounts.find((bankAccount) => bankAccount.plaidAccountID === personalBankAccountDraft?.plaidAccountID);
+
+        if (selectedPlaidBankAccount) {
+            BankAccounts.addPersonalBankAccount(selectedPlaidBankAccount);
+            Navigation.navigate(ROUTES.SETTINGS_ENABLE_PAYMENTS);
+        }
+    }, [personalBankAccountDraft?.plaidAccountID, plaidData?.bankAccounts]);
+
+    const isSetupTypeChosen = userWallet?.setupType === CONST.BANK_ACCOUNT.SETUP_TYPE.PLAID;
+
+    const {componentToRender: SubStep, isEditing, screenIndex, nextScreen, prevScreen, moveTo} = useSubStep({bodyContent: plaidSubsteps, startFrom: 0, onFinished: submit});
+
+    useEffect(() => BankAccounts.clearPersonalBankAccount, []);
+
+    const exitFlow = useCallback(
+        (shouldContinue = false) => {
+            const exitReportID = personalBankAccount?.exitReportID;
+            const onSuccessFallbackRoute = personalBankAccount?.onSuccessFallbackRoute ?? '';
+
+            if (exitReportID) {
+                Navigation.dismissModal(exitReportID);
+            } else if (shouldContinue && onSuccessFallbackRoute) {
+                PaymentMethods.continueSetup(onSuccessFallbackRoute);
+            } else {
+                Navigation.goBack(ROUTES.SETTINGS_WALLET);
+            }
+        },
+        [personalBankAccount],
+    );
+
+    const handleBackButtonPress = () => {
+        switch (true) {
+            case !isSetupTypeChosen:
+                exitFlow();
+                break;
+            case screenIndex === 0:
+                BankAccounts.clearPersonalBankAccount();
+                break;
+            default:
+                prevScreen();
+                break;
+        }
```

Comment:
I; 'm not sure is `switch` method is accurate here. If we don't have a key to compare and we base on logic conditions more accurately is using few `if` instructions  and early return pattern